### ISSUE TITLE
WorldPainter

### ISSUE THREAD

**Issue by Rak1ta** [Botspot/pi-apps#1870]
### Confirmations

- [X] I searched for similar bug reports in https://github.com/Botspot/pi-apps/issues and found none were relevant.
- [X] I tried updating Pi Apps and the issue is still not fixed. [How to update Pi Apps?](https://github.com/Botspot/pi-apps#to-update-pi-apps)

### What happened?

 WorldPainter does not interact with the user

### Description

The application starts and does not interact with the user. Sometimes it turns out to press one of the menu items and everything freezes. Arrow cursor disappears. The application can be minimized, but not closed.

After installation, after a reboot, an identical additional program shortcut appears in the start menu.

![Безымянный](https://user-images.githubusercontent.com/88277343/170078731-cc5faed3-78cc-4ca7-8566-a8faa193fd88.png)
![Безымянный2](https://user-images.githubusercontent.com/88277343/170078738-71b1c711-c006-4bc3-87f3-e0aa048c4272.png)

### What is your operating system?

RaspiOS 32bit Debian10

**Comment by Botspot** [comment]
Pinging @irtexo - the submitter of this app

**Comment by theofficialgman** [comment]
> ### Description
> The application starts and does not interact with the user. Sometimes it turns out to press one of the menu items and everything freezes. Arrow cursor disappears. The application can be minimized, but not closed.

can't reproduce, everything works perfect here... what is your default java version? ```java --version```

> After installation, after a reboot, an identical additional program shortcut appears in the start menu.

the is an upstream bug with the deb makeing its own .desktop file in two places

**Comment by theofficialgman** [comment]
I suggest opening on issue on the official support forum, which is the subreddit
https://www.reddit.com/r/Worldpainter/comments/uosl0c/worldpainter_210_released/

**Comment by Rak1ta** [comment]
@theofficialgman 
```pi@raspberrypi:~ $ java --version
openjdk 16.0.1 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-16.0.1+9 (build 16.0.1+9)
OpenJDK Server VM AdoptOpenJDK-16.0.1+9 (build 16.0.1+9, mixed mode)
```

**Comment by theofficialgman** [comment]
> @theofficialgman
> 
> ```
> openjdk 16.0.1 2021-04-20
> OpenJDK Runtime Environment AdoptOpenJDK-16.0.1+9 (build 16.0.1+9)
> OpenJDK Server VM AdoptOpenJDK-16.0.1+9 (build 16.0.1+9, mixed mode)
> ```

any difference in function if you use your java 11 install?
```
INSTALL4J_JAVA_HOME_OVERRIDE="/usr/lib/jvm/java-11-openjdk-*" /opt/worldpainter/worldpainter
```

**Comment by Rak1ta** [comment]
@theofficialgman no difference, just noticed that the bug works when the cursor is hovering over the map area.

**Comment by theofficialgman** [comment]
I don't really follow what the problem is. can you post a video? its likely not an application issue though. if its graphical, its probably a pi gpu bug

**Comment by Rak1ta** [comment]
@theofficialgman now I can't video anymore, but when the "bug" is triggered, the application cannot scale itself (to full screen or vice versa). Maybe someone from Buster can check it out.

**Comment by Rak1ta** [comment]
@theofficialgman I found what was breaking my application. Tested on a clean install of Buster32bit.

1 - it's just a new clean Buster, WorldPainter is working fine
2- I have installed Minecraft Java MultiMC5. The program no longer works with me.
3 -  Returned the state when WorldPainter is running

![Безымянный](https://user-images.githubusercontent.com/88277343/170301130-5125a5fc-35fc-44d3-b68c-c8aa1e567edd.png)

**Comment by theofficialgman** [comment]
> 1 - it's just a new clean Buster, WorldPainter is working fine 2- I have installed Minecraft Java MultiMC5. The program no longer works with me. 3 - Returned the state when WorldPainter is running

this is not the issue. I already had you test using java-11 using this command: `INSTALL4J_JAVA_HOME_OVERRIDE="/usr/lib/jvm/java-11-openjdk-*" /opt/worldpainter/worldpainter` and you reported no change

**Comment by theofficialgman** [comment]
hmm... seems the * is causing issues
if you run with `INSTALL4J_JAVA_HOME_OVERRIDE="/usr/lib/jvm/java-11-openjdk-armhf" /opt/worldpainter/worldpainter`
you will have no problems regardless of the default java

**Comment by Rak1ta** [comment]
@theofficialgman On the left, java-16 is active, on the right, java-11.
![Безымянный](https://user-images.githubusercontent.com/88277343/170305903-b812a399-1f4c-4a2b-b44b-837527887ba1.png)

**Comment by theofficialgman** [comment]
not really a bug.. it just doesn't work on java 16.

this is a bug though and needs to be fixed upstream:
```
After installation, after a reboot, an identical additional program shortcut appears in the start menu.
```

**Comment by Botspot** [comment]
This is not the first app to force a specific version of Java. But this is the first one I've seen that does it with the `INSTALL4J_JAVA_HOME_OVERRIDE` variable.
Was there an advantage to doing this? Should other apps that set `JAVA_HOME` and `PATH` be changed?

**Comment by theofficialgman** [comment]
> This is not the first app to force a specific version of Java. But this is the first one I've seen that does it with the `INSTALL4J_JAVA_HOME_OVERRIDE` variable. Was there an advantage to doing this? Should other apps that set `JAVA_HOME` and `PATH` be changed?

/opt/worldpainter/worldpainter is a VERY long script to launch worldpainter (you should read it, its a mess).... it messes with a bunch of stuff and internally provides the `INSTALL4J_JAVA_HOME_OVERRIDE` variable. JAVA_HOME and PATH are not used in it

this is a script specific implementation and should not be used elsewhere

**Comment by theofficialgman** [comment]
going to re-open this as there actually isn't any way for me to uninstall that second .desktop file or know where it is... It has to be executed as apt otherwise the location where a normal user (or sudo user) attempts to install is different 😢

**Comment by Rak1ta** [comment]
@theofficialgman I [wrote](https://www.reddit.com/r/Worldpainter/comments/uxlhcq/identical_additional_program_shortcut_appears_in/) to support

**Comment by Rak1ta** [comment]
@theofficialgman The developer suggested a [solution](https://github.com/Captain-Chaos/WorldPainter/issues/258#issuecomment-1140317446) to the problem.
And the extra icon is still not deleted. It's just that there is confusion that the program either starts or it doesn't. 
![Безымянный](https://user-images.githubusercontent.com/88277343/171034362-ff6913b8-6dd9-4be9-9928-ee802df81925.png)

**Comment by theofficialgman** [comment]
> @theofficialgman The developer suggested a [solution](https://github.com/Captain-Chaos/WorldPainter/issues/258#issuecomment-1140317446) to the problem. And the extra icon is still not deleted. It's just that there is confusion that the program either starts or it doesn't. ![Безымянный](https://user-images.githubusercontent.com/88277343/171034362-ff6913b8-6dd9-4be9-9928-ee802df81925.png)

this is not a solution to the duplicate icon. its a solution to needing to override the JRE in the .desktop file. I will implement that

**Comment by Rak1ta** [comment]
@theofficialgman I write incorrectly.  Now in install it is written to remove an extra icon, but it does not work for me.

**Comment by theofficialgman** [comment]
> @theofficialgman I write incorrectly. Now in install it is written to remove an extra icon, but it does not work for me.

thats correct. the change I made does not work. only apt can run the command to uninstall the extra icon
